lapis2-docs/src/content/docs/tutorials/maintainer-tutorials/start-lapis-and-silo.mdx

```diff
@@ -0,0 +1,219 @@
+Note that you need to provide the paths where the files will be stored in the Docker container.
+Filenames are relative to the input directory.
+Since you don't provide the input directory explicitly, SILO will fall back to the default `/data`.
+
+##### `~/lapisExample/config/preprocessing_config.yaml`
+
+```yaml
+metadataFilename: 'small_metadata_set.tsv'
+pangoLineageDefinitionFilename: 'pangolineage_alias.json'
+referenceGenomeFilename: 'reference_genomes.json'
+```
+
+Start the SILO Docker container with the options:
+
+-   start SILO in the preprocessing mode
+-   mount the data into the container to the default location
+-   mount the preprocessing config into the container to the default location
+-   mount the database config into the container to the default location
+-   mount the output directory into the container to the default location
+
+The following command puts it all together:
+
+```
+docker run \
+  --volume ~/lapisExample/data:/preprocessing/input \
+  --volume ~/lapisExample/config/preprocessing_config.yaml:/app/preprocessing_config.yaml \
+  --volume ~/lapisExample/config/database_config.yaml:/app/database_config.yaml \
+  --volume ~/lapisExample/output:/preprocessing/output \
+  ghcr.io/genspectrum/lapis-silo --preprocessing
+```
+
+After this has completed, the output directory should contain the result of the preprocessing.
+This has to be provided to SILO in the next step.
+
+### Starting SILO
+
+Start the SILO Docker container with the options:
+
+-   expose port 8081 to the host.
+-   mount the preprocessing result into the container.
+
+The following command puts it all together:
+
+```
+docker run --detach \
+  --publish 8081:8081 \
+  --volume ~/lapisExample/output:/data \
+  ghcr.io/genspectrum/lapis-silo --api
+```
+
+Now SILO should be available at http://localhost:8081.
+
+### Starting LAPIS
+
+Now you can start LAPIS. You have to:
+
+-   expose port 8080 to the host.
+-   mount the database configuration and the reference genomes to the default locations in the Docker container.
+-   provide LAPIS with the SILO URL.
+
+The following command puts it all together:
+
+```
+docker run --detach \
+  --publish 8080:8080 \
+  --volume ~/lapisExample/config/database_config.yaml:/workspace/database_config.yaml \
+  --volume ~/lapisExample/data/reference_genomes.json:/workspace/reference_genomes.json \
+  ghcr.io/genspectrum/lapis-v2 \
+    --silo.url=http://localhost:8081
+```
+
+Now LAPIS should be available at http://localhost:8080.
+LAPIS offers a Swagger UI that serves as a good starting point for exploring its functionalities.
+You will notice that LAPIS cannot connect to SILO since the address `localhost:8081` is trapped inside the container.
+Luckily Docker Compose comes to the rescue.
+
+## Using Docker Compose
+
+The above `docker run` commands can be combined into a `docker-compose.yaml` file:
+
+##### `~/lapisExample/docker-compose.yaml`
+
+```yaml
+version: '3.9'
```

Comment:
Can we somehow add these files to our codebase and then only show them here? This way we could have a test, which shows this description is also working?

Reply:
That would require writing a test. In principle this should be possible - in the worst case we somehow have to copy the files into the Docker image.
Is it worth the effort? In the end, one would need to automate the whole tutorial.

Reply:
I just thought about testing our documentation, so that it is always up to date. Similar to our e2e tests with the small test sample. But we can tackle this later on (or write a ticket for that)